hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/BlockRecordManager.java

```diff
@@ -45,14 +50,25 @@ public class RecordManager {
      *
      * @param consensusTime The consensus time of the user transaction we are about to start executing. It must be the adjusted consensus time
      *                      not the platform assigned consensus time. Assuming the two are different.
+     * @param state         The state to read BlockInfo from and update when new blocks are created
      */
-    public void startUserTransaction(Instant consensusTime) {}
+    void startUserTransaction(Instant consensusTime, HederaState state);
```

Comment:
So far we tried to avoid passing `HederaState` around. Is it possible to use `WritableStates`, `WritableKVState` or a store instead?

Reply:
The question is who calls `final WritableStates writableStates = state.createWritableStates(BlockRecordService.NAME);` at the moment I have it as BlockRecordManagerImpl it could move to `HandleWorkflow#handlePlatformTransaction()` I am not sure why that is better?

Reply:
I always try to reduce the data that is passed around as much as possible. This helps to avoid modifying unrelated data accidentally.

Also, one has to be careful with what can be done with `state`. It does not see any changes until they are committed. I think this is not a problem right now, because `BlockRecordManager` is probably the only one using `BlockRecordService.NAME` and `state` is not stored. But it is something to be aware of.